force-app/main/default/classes/AccountFieldsSet.cls

```diff
@@ -0,0 +1,25 @@
+public with sharing class AccountFieldsSet {
+    public static void handleInsert(List<Account> newAccounts){
+        List<Account> accountsToBeInserted = new List<Account>();
+        for (Account acc : newAccounts){
+            if (acc.Type == null) {
+                acc.Type = 'Prospect';
+                accountsToBeInserted.add(acc);
+            }
+
+            if (acc.ShippingStreet != null || acc.ShippingCity != null || acc.ShippingState != null || acc.ShippingPostalCode != null || acc.ShippingCountry != null) {
+                acc.BillingStreet = acc.ShippingStreet;
+                acc.BillingCity = acc.ShippingCity;
+                acc.BillingState = acc.ShippingState;
+                acc.BillingPostalCode = acc.ShippingPostalCode;
+                acc.BillingCountry = acc.ShippingCountry;
+                accountsToBeInserted.add(acc);
+            }
```

Comment:
Good another option here would be to use the null coalescing operator 

```
 private void setDefaultBillingAddress(){
        for(Account account : accounts){
            account.BillingStreet = account.ShippingStreet ?? '';
            account.BillingCity = account.ShippingCity ?? '';
            account.BillingState = account.ShippingState ?? '';
            account.BillingPostalCode = account.ShippingPostalCode ?? '';
            account.BillingCountry = account.ShippingCountry ?? '';

        }
    }
    ```

Reply:
This makes more sense. As the code that I wrote, it only updates those account records that are having all fields of the Shipping address as not null but the Null Coalescing operator, updates all the account records with respective shipping address field values if it is not null, if it is null the same will be put in Billing Address.